### PR TITLE
Update various rules

### PR DESCRIPTION
**Pull request by drebrez**
-   Added typescript to the import/resolver
-   Added deprecated React types to the "ban-types" rule
-   Added "complexity" rule with a maximum of 20
-   Added "max-lines" rule with a maximum of 300 lines
-   Updated "no-empty-function" rule to not allow any empty function
-   Updated "quotes" rule by using the typescript version
-   Updated "react/no-unstable-nested-components" rule to allow component creation inside component props

### CODE REVIEW

File: rules/eslint.js [hunk 1]
```diff
@@ -59,9 +71,5 @@ module.exports = {
         message: "`with` is disallowed in strict mode because it makes code impossible to predict and optimize.",
       },
     ],
-
-    // The airbnb config forces single quote, but we prefer double quote
-    // https://eslint.org/docs/latest/rules/quotes
-    quotes: ["error", "double", { avoidEscape: true }],
```
Comment (neotrow) [inline rules/eslint.js:65]: why did you remove this?

File: rules/typescript.js [hunk 2]
```diff
@@ -172,6 +180,11 @@ module.exports = {
     // https://typescript-eslint.io/rules/prefer-readonly-parameter-types
     "@typescript-eslint/prefer-readonly-parameter-types": "off",
 
+    // Enforce the consistent use of double quotes, but allow single quotes so long as the
+    // string contains a quote that would have to be escaped otherwise
+    // https://typescript-eslint.io/rules/quotes
+    "@typescript-eslint/quotes": ["error", "double", { avoidEscape: true }],
```
Comment (neotrow) [inline rules/typescript.js:186]: ah, i see quotes have been moved here

**Review by neotrow — COMMENTED** [review]
what about the other rules that had many false positives like the one complaining about statements that can never be false although they can?

File: rules/eslint.js (hunk 1 above)
Comment (drebrez) [reply rules/eslint.js:65]: this rule in the eslint gets disabled by the typescript-eslint plugin, I didn't removed it, I moved it to the typescript rule

**Review by neotrow — APPROVED** [review]
